crates/sui-types/src/signature.rs

```diff
@@ -4,38 +4,50 @@
 use crate::committee::EpochId;
 use crate::crypto::{SignatureScheme, SuiSignature};
 use crate::zk_login_authenticator::ZkLoginAuthenticator;
+use crate::zk_login_util::OAuthProviderContent;
 use crate::{base_types::SuiAddress, crypto::Signature, error::SuiError, multisig::MultiSig};
 pub use enum_dispatch::enum_dispatch;
 use fastcrypto::{
     error::FastCryptoError,
     traits::{EncodeDecodeBase64, ToFromBytes},
 };
+use im::hashmap::HashMap as ImHashMap;
 use schemars::JsonSchema;
 use serde::Serialize;
 use shared_crypto::intent::IntentMessage;
 use std::hash::Hash;
+use std::sync::Arc;
+
 #[derive(Default, Debug, Clone)]
-pub struct AuxVerifyData {
+pub struct VerifyParams {
     pub epoch: Option<EpochId>,
-    pub google_jwk_as_bytes: Option<Vec<u8>>,
+    //pub google_jwk_as_bytes: Option<Vec<u8>>,
+    //pub oauth_provider_jwk: Option<Arc<OAuthProviderContent>>,
+
+    // map from kid => OauthProviderContent
+    pub oauth_provider_jwks: ImHashMap<String, Arc<OAuthProviderContent>>,
 }
 
-impl AuxVerifyData {
-    pub fn new(epoch: Option<EpochId>, google_jwk_as_bytes: Option<Vec<u8>>) -> Self {
+impl VerifyParams {
+    pub fn new(
+        epoch: Option<EpochId>,
+        oauth_provider_jwks: ImHashMap<String, Arc<OAuthProviderContent>>,
```

Comment:
ooc, why is Arc<OAuthProviderContent> need to be in an arc?

Reply:
Ah, actually, it doesn't anymore, let me revert that